recipes/ccache/all/conanfile.py

```diff
@@ -71,7 +75,11 @@ def validate(self):
             raise ConanInvalidConfiguration("ccache does not support ARMv8 on Windows before version 4.10")
 
     def build_requirements(self):
-        self.tool_requires("cmake/[>=3.15 <4]")
+        # https://ccache.dev/releasenotes.html#_ccache_4_12
+        if Version(self.version) >= "4.12":
+            self.tool_requires("cmake/[>=3.18 <5]")
+        else:
+            self.tool_requires("cmake/[>=3.15 <4]")
```

Comment:
I think it's simpler to just upgrade the CMake version; the idea is to keep the recipe simple

```suggestion
        self.tool_requires("cmake/[>=3.18 <4]")
```

Reply:
Then we can simply remove the upper bound `self.tool_requires("cmake/[>=3.18]")`